src/test/java/com/savvato/tribeapp/services/CosignServiceImplTest.java

```diff
@@ -91,4 +98,88 @@ public void saveCosignAlreadyExisting() {
         assertEquals(cosignDTO.userIdReceiving, expectedCosignDTORepeat.userIdReceiving);
         assertEquals(cosignDTO.phraseId, expectedCosignDTORepeat.phraseId);
     }
+
+    @Test
+    public void testGetCosignersForUser(){
+        // test data
+        Long testUserIdIssuing = 1L;
+        String testUsernameIssuing = "test";
+        Long testUserIdReceiving = 2L;
+        Long testPhraseId = 1L;
+
+        // mock return data
+        UsernameDTO mockUsernameDTO = UsernameDTO.builder()
+                .userId(testUserIdIssuing)
+                .username(testUsernameIssuing)
+                .build();
+
+        List<Long> mockCosignerIds = new ArrayList<>();
+        mockCosignerIds.add(testUserIdIssuing);
+
+        // mock returns
+        when(cosignRepository.findCosignersByUserIdReceivingAndPhraseId(anyLong(),anyLong())).thenReturn(mockCosignerIds);
+        when(userService.getUsernameDTO(anyLong())).thenReturn(mockUsernameDTO);
+
+        // expected results
+        List<UsernameDTO> expectedListUsernameDTO = new ArrayList<>();
+        expectedListUsernameDTO.add(mockUsernameDTO);
+
+        // test
+        List<UsernameDTO> usernameDTOS = cosignService.getCosignersForUserAttribute(testUserIdReceiving,testPhraseId);
+
+        for(UsernameDTO user : usernameDTOS){
+            assertEquals(user.userId, expectedListUsernameDTO.get(0).userId);
+            assertEquals(user.username, expectedListUsernameDTO.get(0).username);
+        }
+    }
+
+    @Test
+    public void testGetAllCosignsForUser() {
```

Comment:
Likewise, create another test, and make the list of expected results three items in length. This test covers the case when there is one element, but not more than one. A test for three items should suffice.

Reply:
I added a test for three phrases and three users per phrase